Microsoft.Toolkit.Uwp.Services/Services/LinkedIn/LinkedInDataProvider.cs

```diff
@@ -0,0 +1,317 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Services.Exceptions;
+using Windows.Data.Json;
+using Windows.Security.Authentication.Web;
+using Windows.Security.Credentials;
+using Windows.Storage;
+using Windows.Web.Http;
+
+namespace Microsoft.Toolkit.Uwp.Services.LinkedIn
+{
+    /// <summary>
+    /// Data Provider for connecting to LinkedIn service.
+    /// </summary>
+    public class LinkedInDataProvider
+    {
+        private const string OAuthBaseUrl = "https://www.linkedin.com/uas/oauth2/";
+        private const string BaseUrl = "https://api.linkedin.com/v1";
+
+        /// <summary>
+        /// Tokens for service.
+        /// </summary>
+        private readonly LinkedInOAuthTokens tokens;
+
+        /// <summary>
+        /// Password vault used to store access tokens
+        /// </summary>
+        private readonly PasswordVault vault;
+
+        /// <summary>
+        /// Gets or sets logged in user information.
+        /// </summary>
+        public string Username { get; set; }
+
+        /// <summary>
+        /// Gets a value indicating whether the provider is already logged in
+        /// </summary>
+        public bool LoggedIn { get; private set; }
+
+        /// <summary>
+        /// Gets or sets requiredPermissions.
+        /// </summary>
+        public LinkedInPermissions RequiredPermissions { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LinkedInDataProvider"/> class.
+        /// Constructor.
+        /// </summary>
+        /// <param name="tokens">OAuth tokens for request.</param>
+        /// <param name="requiredPermissions">Required permissions for the session.</param>
+        public LinkedInDataProvider(LinkedInOAuthTokens tokens, LinkedInPermissions requiredPermissions)
```

Comment:
For consistency sake, I guess we should have two intiliaze methods like in Facebook service: https://github.com/Microsoft/UWPCommunityToolkit/blob/dev/Microsoft.Toolkit.Uwp.Services/Services/Facebook/FacebookService.cs#L85